apps/ecommerce/lambda/src/middlewares/getAppInstallationParameters.ts

```diff
@@ -0,0 +1,86 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express';
+import { getManagementToken } from '@contentful/node-apps-toolkit';
+import axios from 'axios';
+import { UnableToGetAppInstallationParameters } from '../errors/unableToGetAppInstallationParameters';
+import { AppInstallationParameters } from '../types';
+import { UnableToGetAppAccessToken } from '../errors/unableToGetAppAccessToken';
+
+export const getAppInstallationParametersMiddleware: RequestHandler = async (
+  req: Request,
+  _res: Response,
+  next: NextFunction
+) => {
+  try {
+    const installationParameters: AppInstallationParameters = {};
+
+    // installation parameters are provided via x-data-provider-parameters header on GET requests
+    // to check credentials prior to app installation / saving of app installation parameters
+    const parametersHeader = req.header('x-data-provider-parameters');
+
+    if (parametersHeader) {
+      Object.assign(installationParameters, JSON.parse(parametersHeader));
+    } else {
+      const appId = req.header('x-contentful-app');
+      const spaceId = req.header('x-contentful-space-id');
+      const environmentId = req.header('x-contentful-environment-id');
+
+      if (!appId || !spaceId || !environmentId) {
+        console.log('headers', req.headers);
+        throw new Error('Missing required headers!');
+      }
+
+      if (!req.appConfig) throw new Error('App config not found');
+
+      const appAccessToken = await getManagementToken(req.appConfig.privateKey, {
```

Comment:
It could just be, but mixing await with then here really does my head in, and also leads us to awkward things like `then((token) => token)` This could all be refactored to:

```js
let appAccessToken
try {
  appAccessToken = await getManagementToken(req.appConfig.privateKey, {
        appInstallationId: appId,
        spaceId,
        environmentId,
      })
} catch(e) {
  throw new UnableToGetAppAccessToken(e.message, {
    cause: e,
  });
}
```